govuk-one-login/authentication-frontend
# BAU: Clean up after cloudfront implementation

## What

Some resources and feature flags were left over after the cloudfront integration. As we now use cloudfront in all environments, these switches and old configuration are no longer required.

## How to review

1. `git checkout main`
2. `deploy-xx.sh -t --shell`
3. `terraform apply **command line options echoed out, with -auto-approve removed**`
4. Ensure it's applied successfully.
5. RUN THE FOLLOWING COMMANDS IN THE SAME TERMINAL SESSION, IMMEDIATELY AFTER THE LAST `apply`
6. `git checkout BAU/clean-up-cloudfront`
7. `terraform apply **command line options echoed out, with -auto-approve removed**` (the exact same command as you ran in step 3)
8. Ensure that _only_ the following are marked for destruction:
   - `aws_wafv2_web_acl_logging_configuration.frontend_alb_waf_logging_config`
   - `aws_wafv2_web_acl.frontend_alb_waf_regional_web_acl`
   - `aws_wafv2_ip_set.gds_ip_set`
9. Ensure that all other changes are only resources being `moved`.

If there are any other spurious changes:

1. `git checkout main`
2. `git pull`
3. `git checkout BAU/clean-up-cloudfront`
4. `git rebase main`
5. rerun the above steps
6. if they pass this time, `git checkout BAU/clean-up-cloudfront && git push --force-with-lease` before approving the PR.
